### PR TITLE
OP - Add all ERC20 tokens to erc20.tokens

### PR DESCRIPTION
**Pull request by MSilb7**
Brief comments on the purpose of your changes:

~Note: This should only run after https://github.com/duneanalytics/spellbook/pull/3158~

Make tokens_optimism.erc20 match the structure of tokens.nft , where all tokens are identified via transfers, but only curated/known tokens have symbols.

This structure also lets up pull in "algorithmically-generated" tokens (i.e. aave atoken factory) - which we can expand to other protocols (given no contract read solution).

*Q: Is it possible from here to run contract reads on unmapped tokens to fill in symbols and decimals? This would remove all admin work of curating token lists. This is the ideal solution imo.*

For generating the broader tokens.erc20 table, only known tokens are shown.

---

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by MSilb7** [comment]
Made an edit for the Aave and Granary tokens file. The error seems to be from not recognizing the jinja.

**Comment by MSilb7** [comment]
bumping - the error seems to be from the test not recognizing dynamic jinja (this part was unchanged).

But also relevant if this is how erc20 tables should be represented (to match nft version). I think it's helpful for tokens table to include all known tokens, then a few are curated or generated. cc @jeff-dude

File: models/tokens/optimism/tokens_optimism_erc20_generated.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{{ config( alias='erc20_generated', tags=['static'],
```
Comment (jeff-dude) [inline models/tokens/optimism/tokens_optimism_erc20_generated.sql:1]: seems this one shouldn't be tagged static?

File: models/yearn/optimism/yearn_optimism_vaults.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{{ config(
+        schema = 'yearn'
+        , alias = 'vaults'
+        , materialized = 'table'
```
Comment (jeff-dude) [inline models/yearn/optimism/yearn_optimism_vaults.sql:4]: seems this needs to be incremental

File: models/tokens/optimism/tokens_optimism_erc20_generated.sql (hunk 1 above)
Comment (MSilb7) [reply models/tokens/optimism/tokens_optimism_erc20_generated.sql]: ah yeah copy and paste error probably

File: models/yearn/optimism/yearn_optimism_vaults.sql (hunk 2 above)
Comment (MSilb7) [reply models/yearn/optimism/yearn_optimism_vaults.sql]: good catch!

**Comment by MSilb7** [comment]
> 

Yeah agree, it will be a weird thing whenever these Aave tables are hit.

Made the table info edits!

**Review by jeff-dude — APPROVED** [review]
should be all good

**Pull request merged**